contribs/gnogenesis/internal/txs/txs_add_packages.go

```diff
@@ -65,30 +82,46 @@ func execTxsAddPackages(
 	io commands.IO,
 	args []string,
 ) error {
+	var (
+		keyname = DefaultAccount_Name
+		keybase keys.Keybase
+		pass    string
+	)
 	// Load the genesis
-	genesis, loadErr := types.GenesisDocFromFile(cfg.txsCfg.GenesisPath)
-	if loadErr != nil {
-		return fmt.Errorf("unable to load genesis, %w", loadErr)
+	genesis, err := types.GenesisDocFromFile(cfg.txsCfg.GenesisPath)
+	if err != nil {
+		return fmt.Errorf("unable to load genesis, %w", err)
 	}
 
 	// Make sure the package dir is set
 	if len(args) == 0 {
 		return errInvalidPackageDir
 	}
 
-	var (
-		creator = defaultCreator
-		err     error
-	)
-
-	// Check if the deployer address is set
-	if cfg.deployerAddress != defaultCreator.String() {
-		creator, err = crypto.AddressFromString(cfg.deployerAddress)
+	if cfg.keyName != "" {
+		keyname = cfg.keyName
+		keybase, err = keys.NewKeyBaseFromDir(cfg.gnoHome)
+		if err != nil {
+			return fmt.Errorf("unable to load keybase: %w", err)
+		}
+		pass, err = io.GetPassword("Enter password.", cfg.insecurePasswordStdin)
+		if err != nil {
+			return fmt.Errorf("cannot read password: %w", err)
+		}
+	} else {
+		keybase = keys.NewInMemory()
+		_, err := keybase.CreateAccount(DefaultAccount_Name, DefaultAccount_Seed, "", "", 0, 0)
 		if err != nil {
-			return fmt.Errorf("%w, %w", errInvalidDeployerAddr, err)
+			return fmt.Errorf("unable to create account: %w", err)
 		}
 	}
 
+	info, err := keybase.GetByNameOrAddress(keyname)
+	if err != nil {
+		return err
```

Comment:
We should wrap the error, ex `fmt.Errorf("unable to find key in keybase: %w", err)`

Reply:
done 2a67769b4f165ffd4d9ed62cf27891387fa8b949